taichi/transforms/simplify.cpp

```diff
@@ -324,6 +351,35 @@ class BasicBlockSimplify : public IRVisitor {
     set_done(stmt);
   }
 
+  bool has_load(int start_index, int end_index, LocalStoreStmt *stmt) {
+    for (int j = start_index; j <= end_index; j++) {
+      std::function<bool(Stmt *)> check_load;
+      check_load = [&] (Stmt *stmt_j) {
+        if (stmt_j->is_container_statement()) {
+          // if, while, etc..
+          return true; // TODO
+        }
+        if (stmt_j->is<LocalLoadStmt>() &&
+            stmt_j->as<LocalLoadStmt>()->has_source(stmt->ptr))
+          return true;
+        if (stmt_j->is<AtomicOpStmt>() &&
+            (stmt_j->as<AtomicOpStmt>()->dest ==
+                stmt->ptr)) {
+          // $a = alloca
+          // $b : local store [$a <- v1]  <-- prev lstore |bstmt_|
+          // $c = atomic add($a, v2)      <-- cannot eliminate $b
+          // $d : local store [$a <- v3]
+          return true;
+        }
+        return false;
+      };
+      if (!StmtSearcher<Stmt>::run(block->statements[j].get(), check_load).empty()) {
```

Comment:
Make sure you safeguard your modifications with that global var so that you can easily benchmark later :-)